test/api/controllers/routerController/initRouterHttp.test.js

```diff
@@ -29,11 +30,22 @@ function parseHttpResponse(response) {
   response.on('end', function () {
     var result;
 
-    try {
-      result = JSON.parse(data);
-    }
-    catch (e) {
-      deferred.reject(e);
+    if (yaml === true) {
+      yamlToJson(data, (err, res) => {
+        if (err) {
+          deferred.reject(err);
+        }
+
+        result = res;
+      });
+    } 
+    else {
+      try {
+        result = JSON.parse(data);
+      }
+      catch (e) {
+        deferred.reject(e);
```

Comment:
since the execution flow isn't stopped, the deferred promise will eventually be resolved instead
you should directly returns this with `return deferred.reject(err);`